diff --git a/src/Framework/TestResult.php b/src/Framework/TestResult.php
index 4fde29fac..6be437905 100644
--- a/src/Framework/TestResult.php
+++ b/src/Framework/TestResult.php
@@ -37,6 +37,8 @@
 use SebastianBergmann\ResourceOperations\ResourceOperations;
 use SebastianBergmann\Timer\Timer;
 use Throwable;
+use AKEI\TimeLogger;
+
 
 /**
  * @internal This class is not covered by the backward compatibility promise for PHPUnit
@@ -693,6 +695,8 @@ function_exists('xdebug_start_function_monitor');
             xdebug_start_function_monitor(ResourceOperations::getFunctions());
         }
 
+
+        TimeLogger::resetTimer();
         $timer = new Timer;
         $timer->start();
 
@@ -772,7 +776,8 @@ function_exists('xdebug_start_function_monitor');
             $error = true;
         }
 
-        $time = $timer->stop()->asSeconds();
+
+        $time = TimeLogger::getExecutionTime() ?? $timer->stop()->asSeconds();
 
         $test->addToAssertionCount(Assert::getCount());
 
